extensions/wikia/SpecialDiscussions/SpecialDiscussionsController.class.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+use Swagger\Client\ApiException;
+use Swagger\Client\Discussion\Api\SitesApi;
+use Wikia\DependencyInjection\Injector;
+use Wikia\Service\Swagger\ApiProvider;
+
+/**
+ * Discussion user log page
+ */
+class SpecialDiscussionsController extends WikiaSpecialPageController {
+
+	const HTTP_STATUS_OK = 200;
+	const DISCUSSIONS_ACTION = 'specialdiscussions';
+	const SERVICE_NAME = 'discussion';
+	const TIMEOUT = 5;
+	const EDIT_TOKEN = 'editToken';
+
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+
+	private $logger;
+	private $sitesApi;
+	private $siteId;
+	private $siteName;
+
+	public function __construct() {
+		parent::__construct( 'Discussions', '', false );
+		$this->sitesApi = $this->getDiscussionSitesApi();
+		$this->siteId = F::app()->wg->CityId;
+		$this->siteName = WikiFactory::getVarValueByName( 'wgSitename', $this->siteId );
+		$this->logger = Wikia\Logger\WikiaLogger::instance();
+	}
+
+	public function index() {
+		$this->checkAccess();
+
+		$this->setHeaders();
+		$this->response->addAsset( 'special_discussions_scss' );
+
+		$this->wg->Out->setPageTitle( wfMessage( 'discussions-pagetitle' )->escaped() );
+
+		$editToken = $this->getVal( self::EDIT_TOKEN );
+		if ( !empty( $editToken ) ) {
+			$this->activateDiscussions();
+		}
+
+		$this->setIndexOutput();
+	}
+
+	public function inputForm() {
+		$this->checkAccess();
+
+		$this->response->setValues(
+			[
+				'discussionsInactiveMessage' => wfMessage( 'discussions-not-active' )->escaped(),
+				'activateDiscussions' => wfMessage( 'discussions-activate' )->escaped(),
+				'editToken' => $this->getUser()->getEditToken(),
+			]
+		);
+	}
+
+	public function discussionsLink() {
+		$this->checkAccess();
+
+		$this->response->setValues(
+			[
+				'discussionsActiveMessage' => wfMessage( 'discussions-active' )->escaped(),
+				'discussionsLink' => $this->getDiscussionsLink(),
+				'discussionsLinkCaption' => wfMessage( 'discussions-navigate' )->escaped(),
+			]
+		);
+	}
+
+	private function setIndexOutput() {
+		if ( $this->getIsDiscussionsActive() ) {
+			$this->response->setVal(
+				'discussionsLink',
+				$this->sendSelfRequest(
+					'discussionsLink',
+					[
+						'discussionsLink' => $this->getDiscussionsLink(),
+					]
+				)
+			);
+		} else {
+			$this->response->setVal(
+				'inputForm',
+				$this->sendSelfRequest(
+					'inputForm',
+					[
+						'activateDiscussions' => $this->getActivateDiscussionsMessage(),
+					]
+				)
+			);
+		}
+	}
+
+	private function activateDiscussions() {
+		$this->assertValidPostRequest( $this->request, $this->getUser() );
```

Comment:
I think this is more valuable as an instance/stateful method that knows its a SpecialDiscusionController request.  Otherwise, you'd use or extend `WebRequest::isValidWriteRequest`.